lib/preload/ndb/preload.js

```diff
@@ -5,48 +5,47 @@
  */
 
 try {
-  const inspector = require('inspector');
-  const fs = require('fs');
-  let inspectorPort = 0;
-  inspector.open(inspectorPort, undefined, false);
-  let inspectorUrl = inspector.url();
-  if (process.version.startsWith('v8.')) {
-    const url = require('url');
-    const parsedInspectorUrl = url.parse(inspectorUrl);
-    parsedInspectorUrl.pathname = '/json';
-    parsedInspectorUrl.protocol = 'http';
-    inspectorUrl = url.format(parsedInspectorUrl);
-    inspectorPort = Number(parsedInspectorUrl.port);
+  let scriptName = '';
+  try {
+    scriptName = require.resolve(process.argv[1]);
+  } catch (e) {
   }
-  const sep = process.platform === 'win32' ? '\\' : '/';
-  const fileName = `${process.env.NDD_STORE}${sep}${process.pid}`;
-  process.once('exit', fs.unlinkSync.bind(null, fileName));
-  const info = fetchProcessInfo();
-  info.inspectorUrl = inspectorUrl;
-  fs.writeFileSync(fileName, JSON.stringify(info));
-  if (process.version.startsWith('v8.')) {
-    inspector.close();
-    inspector.open(inspectorPort, undefined, true);
-  } else {
-    inspector.open(inspectorPort, undefined, true);
-  }
-
-  function fetchProcessInfo() {
-    let scriptName = '';
-    try {
-      scriptName = require.resolve(process.argv[1]);
-    } catch (e) {
-    }
-    const ppid = process.env.NDD_PPID;
-    process.env.NDD_PPID = process.pid;
-    process.versions['ndb'] = process.env.NDB_VERSION;
-    return {
-      cwd: process.cwd(),
-      argv: process.argv.concat(process.execArgv),
-      data: process.env.NDD_DATA,
-      ppid: ppid,
-      scriptName: scriptName
-    };
+  const ppid = process.env.NDD_PPID;
+  process.env.NDD_PPID = process.pid;
+  if (!process.env.NDD_DATA)
+    process.env.NDD_DATA = process.pid + '_ndbId';
+  process.versions['ndb'] = process.env.NDB_VERSION;
+  const info = {
+    cwd: process.cwd(),
+    argv: process.argv.concat(process.execArgv),
+    data: process.env.NDD_DATA,
+    ppid: ppid,
+    id: String(process.pid),
+    inspectorUrl: require('inspector').url(),
+    scriptName: scriptName
+  };
+  const {execFileSync} = require('child_process');
+  const data = JSON.stringify(info);
+  const code = `(${publish.toString()})(${data}, '${process.env.NDD_IPC}')`;
+  execFileSync(process.execPath, ['-e', `eval(Buffer.from('${Buffer.from(code).toString('base64')}', 'base64').toString())`], {
```

Comment:
This code is hard to understand. I wonder if it could be simplified by running this same file and then switching to a different mode based on command flags.

E.g. take a look at example [here](https://nodejs.org/api/worker_threads.html), where the same script is used to start a worker and as a worker itself.

Reply:
done